gobblin-data-management/src/main/java/org/apache/gobblin/data/management/retention/profile/MultiDatasetFinder.java

```diff
@@ -100,13 +103,23 @@ public MultiDatasetFinder(FileSystem fs, Properties jobProps) {
           importedBys.addAll(client.getImportedBy(new URI(tag), false));
         }
 
+        FileSystem targetFs;
+        if(this.jobProps.containsKey(ConfigurationKeys.WRITER_FILE_SYSTEM_URI)) {
```

Comment:
Why do we need this check? WriterUtils#getWriterFs() already checks for the key. If your intent is to change the default from LOCAL_FS_URI to the local HDFS URI, then you can implement a new getWriterFs() method in WriterUtils that also takes default FS URI as an argument.

Reply:
Why not initialize targetFs in DatasetCleanerJob? Currently, DatasetCleanerJob gets the local HDFS config. You can apply your change before instantiating a DatasetCleaner.

Reply:
Initializing targetFs and passing it to DatasetCleaner instead of fs (SourceFS) will prevent reading the configuration from the local HDFS.